pkg/process/handler.go

```diff
@@ -0,0 +1,174 @@
+package process
+
+import (
+	"github.com/AsterNighT/software-engineering-backend/api"
+	"github.com/labstack/echo/v4"
+	"net/http"
+)
+
+type RegistrationHandler struct {
+}
+
+// GetAllDepartments
+// @Summary get all departments
+// @Tags patient
+// @Description display all departments of a hospital
+// @Produce json
+// @Success 200 {object} api.ReturnedData{data=[]Department}
+// @Router /departments [GET]
+func (h *RegistrationHandler) GetAllDepartments(c echo.Context) error {
+
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+// GetDepartmentByID
+// @Summary get a department by its ID
+// @Tags patient
+// @Description return a department's details by its ID
+// @Param DepartmentID path uint true "department ID"
+// @Produce json
+// @Success 200 {object} api.ReturnedData{data=Department}
+// @Router /departments/{DepartmentID} [GET]
+func (h *RegistrationHandler) GetDepartmentByID(c echo.Context) error {
+
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+// CreateRegistration
+// @Summary create registration
+// @Tags both doctor & patient
+// @Description return registration state
+// @Param PatientID body uint true "patient's ID"
+// @Param DepartmentID body uint true "department ID"
+// @Param RegistrationTime body string true "registration's time, basic format 'YYYY-MM-DD-<half-day>'"
+// @Produce json
+// @Success 200 {object} api.ReturnedData{}
+// @Router /registration [POST]
+func (h *RegistrationHandler) CreateRegistration(c echo.Context) error {
+
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+// NOTE: we assume a doctor may help the patient commit registration
+
+
+
+// GetRegistrationsByPatient
+// @Summary get all registrations (patient view)
+// @Tags patient
+// @Description display all registrations of a patient
+// @Produce json
+// @Success 200 {object} api.ReturnedData{data=[]Registration}
+// @Router /patient/registrations [GET]
+func (h *RegistrationHandler) GetRegistrationsByPatient(c echo.Context) error {
+
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+// NOTE: use cookie for identification
+
+// GetRegistrationsByDoctor
+// @Summary get all registrations (doctor view)
+// @Tags doctor
+// @Description display all registrations of a patient
+// @Param RegistrationID path uint true "registration's ID"
+// @Produce json
+// @Success 200 {object} api.ReturnedData{data=[]Registration}
+// @Router /doctor/registrations [GET]
```

Comment:
```suggestion
// @Router /doctor/registrations/{RegistrationID} [GET]
```

Reply:
sorry, line 76 is an error. `GetRegistrationsByDoctor` should return all registrations assigned to the doctor, not return certain Registration by RegistrationID